modules/auxiliary/cloud/aws/enum_ec2.rb

```diff
@@ -61,15 +66,60 @@ def describe_ec2_instance(i)
     print_good "    Creation Date:  #{i.launch_time}"
     print_good "    Public IP:      #{i.public_ip_address} (#{i.public_dns_name})"
     print_good "    Private IP:     #{i.public_ip_address} (#{i.private_dns_name})"
+    # Report hosts and info
+    mac_addr = i.network_interfaces.select { |iface|
+      iface.private_ip_address == i.private_ip_address
+    }.first.mac_address
+    iname = i.tags.find {|t| t.key == 'Name'} ? i.tags.find {|t| t.key == 'Name'}.value : i.private_dns_name
+    iinfo = i.tags.find {|t| t.key == 'Description'} ? i.tags.find {|t| t.key == 'Description'}.value : nil
+    report_host(
+      host: i.private_ip_address,
+      mac: mac_addr,
+      os_name: i.platform_details,
+      os_flavor: i.architecture,
+      name: iname,
+      info: iinfo,
+      comments: "ec2-id: #{i.id}"
+    )
+    report_note(
+      host: i.private_ip_address,
+      type: 'ec2.public_ip',
+      data: i.public_ip_address
+    ) if i.public_ip_address
+    report_note(
+      host: i.private_ip_address,
+      type: 'ec2.public_dns',
+      data: i.public_dns_name
+    ) if i.public_ip_address and not i.public_dns_name.empty?
+    report_note(
+      host: i.private_ip_address,
+      type: 'ec2.hypervisor',
+      data: i.hypervisor
+    ) if i.hypervisor
+
     i.security_groups.each do |s|
       print_good "    Security Group: #{s.group_id}"
+      report_note(
+        host: i.private_ip_address,
+        type: "ec2.#{s.group_id}",
+        data: s.group_name
+      )
+    end
+
+    i.tags.each do |t|
+      print_good "    Tag: #{t.key} = #{t.value}"
+      report_note(
+        host: i.private_ip_address,
+        type: "ec2.tag #{t.key}",
+        data: t.value
+      )
     end
   end
 
   def run
-    regions = datastore['REGION'] ? [datastore['REGION']] : regions = enumerate_regions()
+    regions = enumerate_regions
```

Comment:
`datastore['REGION']` is still defined should this respect that value still?

Reply:
It does, the logic just moved to the first line of `#enumerate_regions` to `return [datastore['REGION']] if datastore['REGION'] `